drivers/hmis/app/models/hmis/ce/match/candidate.rb

```diff
@@ -0,0 +1,41 @@
+# an eligible, prioritized client for a given candidate pool
+module Hmis::Ce::Match
+  class Candidate < GrdaWarehouseBase
+    self.table_name = 'ce_match_candidates'
+    belongs_to :candidate_pool, class_name: 'Hmis::Ce::Match::CandidatePool'
+    belongs_to :client, class_name: 'Hmis::Hud::Client'
+
+    # FIXME: permissions
+    scope :viewable_by, ->(_user) { all }
+
+    # order by descending priority, NULL values last. Use id as a tie-breaker
+    scope :prioritized, -> {
+      order(
+        Arel::Nodes::Descending.new(arel_table[:priority_score]).nulls_last,
+        arel_table[:id].asc,
+      )
+    }
+
+    # Which candidates (clients) are eligible for an opportunity
+    # - Filter out candidates that have already been referred to this opportunity.
+    # - Filter out candidates with an active referral to another opportunity with overlapping categories.
+    scope :for_opportunity, ->(opportunity) {
+      return Hmis::Ce::Opportunity.none unless opportunity.candidate_pool
+
+      scope = opportunity.candidate_pool.candidates
+
+      # do we need to allow a referral to be re-started for the same client/opportunity?
```

Comment:
We probably need to allow a re-referral.  (Someone starts with an opportunity, disappears, the referral is closed, someone else get referred, fails, and the original person shows up again with the highest priority)